### PR TITLE
remove unneeded cni installation

### PR DESCRIPTION
**Pull request by ndbaker1**
**Issue #, if available:**

**Description of changes:**

if customer's will install 3rd party CNI do we need to continue installing this?

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Testing Done**

*[See this guide for recommended testing for PRs.](../doc/CONTRIBUTING.md#testing-changes) Some tests may not apply. Completing tests and providing additional validation steps are not required, but it is recommended and may reduce review time and time to merge.*

### CODE REVIEW

**Comment by ndbaker1** [comment]
will this break things that log-analyzer is looking for?

**Comment by bryantbiggs** [comment]
will this increase node startup latency; if so, do we know by how much? I assume VPC CNI will have to pull these upon instance creation

**Comment by cartermckinnon** [comment]
> if customer's will install 3rd party CNI do we need to continue installing this?

Generally I would say no. If a CNI implementation is utilizing these binaries, it should include them in its distribution. Calico does this as well: https://github.com/projectcalico/calico/tree/d7934529b7f233c93c93b3ef9efc0ecc55de2a3d/cni-plugin

File: templates/al2023/template.json [hunk 1]
```diff
@@ -183,7 +182,6 @@
         "KUBERNETES_BUILD_DATE={{user `kubernetes_build_date`}}",
         "KUBERNETES_VERSION={{user `kubernetes_version`}}",
         "PAUSE_CONTAINER_VERSION={{user `pause_container_version`}}",
-        "PULL_CNI_FROM_GITHUB={{user `pull_cni_from_github`}}",
```
Comment (cartermckinnon) [inline templates/al2023/template.json:186]: We need to get rid of `CNI_PLUGIN_VERSION` and the associated template var as well

Comment (ndbaker1) [reply templates/al2023/template.json:186]: ack, done

**Comment by ndbaker1** [comment]
Is this rollable into to AL2, or should this just be an AL2023 change since we don't know who might depend on this?

**Comment by cartermckinnon** [comment]
We'll leave this as-is in AL2 👍

**Review by cartermckinnon — APPROVED** [review]

**Pull request merged**